runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/ExecutableStage.java

```diff
@@ -100,9 +109,9 @@
    * </ul>
    *
    * <p>The executable stage can be reconstructed from the resulting {@link ExecutableStagePayload}
-   * and components alone via {@link #fromPayload(ExecutableStagePayload, Components)}.
+   * and components alone via {@link #fromPayload(ExecutableStagePayload)}.
    */
-  default PTransform toPTransform() {
+  default PTransform toPTransform(RunnerApi.Components components) {
```

Comment:
If we now capture the required (perhaps non-strict) subset of components in `getComponents()`, why not call that internally?

Reply:
Yep, rm'd